### PR TITLE
feat: add support for public IP

### LINKED ISSUE #368: Add support for Public IP
### Feature Description

Add support to connect to AlloyDB instances using Public IP

### PR DESCRIPTION
**Pull request by ttosta-google**
Fixes #368

### CODE REVIEW

File: alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/IpType.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.alloydb;
+
+public enum IpType {
+  PUBLIC,
+  PRIVATE,
+  PSC;
```
Comment (jackwotherspoon) [inline alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/IpType.java:21]: We are not adding PSC support to the connectors just yet I believe. I think we want to wait on this until we have tested appropriately...

File: alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultConnectionInfoRepository.java [hunk 2]
```diff
@@ -78,6 +78,7 @@ public ListenableFuture<ConnectionInfo> getConnectionInfo(
 
               return new ConnectionInfo(
                   info.getIpAddress(),
+                  info.getPublicIpAddress(),
```
Comment (jackwotherspoon) [inline alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultConnectionInfoRepository.java:81]: What if an instance does not have a Public IP, does this fail nicely...?

File: alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/IpType.java (hunk 1 above)
Comment (ttosta-google) [reply alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/IpType.java]: updated

File: alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultConnectionInfoRepository.java (hunk 2 above)
Comment (ttosta-google) [reply alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultConnectionInfoRepository.java:81]: Now it raises an RuntimeException when IP address is not provided. Thanks for you review.

**Review by enocom — APPROVED** [review]
Looks great!

**Pull request merged**